code/game/objects/items/props/helmetgarb.dm

```diff
@@ -94,110 +116,342 @@
+			else
+				to_chat(user, SPAN_WARNING("You were interrupted."))
+				break
+
+	else if(HAS_TRAIT(A, TRAIT_TOOL_SCREWDRIVER))
+		if(user.action_busy)
+			return
+		if(src != user.get_inactive_hand())
+			to_chat(user, SPAN_WARNING("You need to hold \the [src] in hand in order to repair them."))
+			return
+		if(!skillcheck(user, SKILL_ENGINEER, SKILL_ENGINEER_TRAINED)) // level 2 is enough to repair damaged NVG
+			to_chat(user, SPAN_WARNING("You are not trained to repair electronics..."))
+			return
+
+		if(shape == NVG_SHAPE_BROKEN)
+			if(!skillcheck(user, SKILL_ENGINEER, SKILL_ENGINEER_ENGI)) // level 3 is needed to repair broken NVG
+				to_chat(user, SPAN_WARNING("Repair of this complexity is too difficult for you, find someone more trained."))
+				return
+
+			to_chat(user, "You begin to repair \the [src].")
+			if(do_after(user, 60, INTERRUPT_ALL, BUSY_ICON_BUILD, src))
+				playsound(src.loc, 'sound/items/Screwdriver.ogg', 25, 1)
+				src.color = "#bebebe"
+				shape = NVG_SHAPE_PATCHED
+				to_chat(user, "You successfully patch \the [src].")
+				nvg_maxhealth = 65
+				nvg_health = 65
+				nvg_drain = initial(nvg_drain) * 2
+			else
+				to_chat(user, SPAN_WARNING("You were interrupted."))
+
+		else if(nvg_health == nvg_maxhealth)
+			if(shape == NVG_SHAPE_PATCHED)
+				to_chat(user, SPAN_WARNING("Already repaired, nothing more you can do."))
+			else if(shape == NVG_SHAPE_FINE)
+				to_chat(user, SPAN_WARNING("Nothing to fix."))
+			else if(shape == NVG_SHAPE_COSMETIC)
+				to_chat(user, SPAN_WARNING("it's nothing but a husk of what it used to be."))
+
+		else
+			to_chat(user, "You begin to repair \the [src].")
+			if(do_after(user, 40, INTERRUPT_ALL, BUSY_ICON_BUILD, src))
+				to_chat(user, "You successfully repair \the [src].")
+				playsound(src.loc, 'sound/items/Screwdriver.ogg', 25, 1)
+				nvg_health = nvg_maxhealth
+			else
+				to_chat(user, SPAN_WARNING("You were interrupted."))
 
-	if(ismob(S.master_object.loc))
-		activation.give_to(S.master_object.loc)
+	else
+		..()
+
+/obj/item/prop/helmetgarb/helmet_nvg/examine(mob/user)
+	. = ..()
+
+	if(shape == NVG_SHAPE_BROKEN)
+		to_chat(user, "They appear to be broken. Maybe someone competent can fix them.")
+	else
+
+		if(shape == NVG_SHAPE_PATCHED)
+			to_chat(user, "They are covered in scratches and have traces of a recent repair.")
+
+		var/nvg_health_procent = nvg_health / nvg_maxhealth * 100
+		if(nvg_health_procent > 70)
+			to_chat(user, "They appear to be in good shape.")
+		else if(nvg_health_procent > 50)
+			to_chat(user, "They are visibly damaged.")
+		else if(nvg_health_procent > 30)
+			to_chat(user, "It's unlikely they can sustain more damage.")
+		else if(nvg_health_procent >= 0)
+			to_chat(user, "They are falling apart.")
+
+	if (get_dist(user, src) <= 1 && (shape == NVG_SHAPE_FINE || shape == NVG_SHAPE_PATCHED))
+		to_chat(user, "A small gauge in the corner reads: Power: [round(100.0*nvg_charge/nvg_maxcharge) ]%.")
```

Comment:
Separate these out into different procs.
Your do_afters should use the SECONDS defines as well